src/python_testing/TC_EWATERHTR_2_2.py

```diff
@@ -0,0 +1,376 @@
+        asserts.assert_not_equal(
+            heatDemand & Clusters.WaterHeaterManagement.Bitmaps.WaterHeaterDemandBitmap.kImmersionElement1, 0)
+
+        self.step("8b")
+        await self.check_whm_attribute("BoostState", Clusters.WaterHeaterManagement.Enums.BoostStateEnum.kActive)
+
+        self.step("9")
+        time.sleep(6)
+
+        self.step("9a")
+        heatDemand = await self.read_whm_attribute_expect_success(attribute="HeatDemand")
+        asserts.assert_equal(heatDemand, 0)
+
+        self.step("9b")
+        await self.check_whm_attribute("BoostState", Clusters.WaterHeaterManagement.Enums.BoostStateEnum.kInactive)
+
+        self.step("10")
+        await self.send_boost_command(duration=600, one_shot=True)
+
+        self.step("10a")
+        heatDemand = await self.read_whm_attribute_expect_success(attribute="HeatDemand")
+        asserts.assert_not_equal(
+            heatDemand & Clusters.WaterHeaterManagement.Bitmaps.WaterHeaterDemandBitmap.kImmersionElement1, 0)
+
+        self.step("10b")
+        await self.check_whm_attribute("BoostState", Clusters.WaterHeaterManagement.Enums.BoostStateEnum.kActive)
+
+        self.step("11")
+        await self.send_test_event_trigger_water_temperature61C_test_event()
+
+        self.step("11a")
+        heatDemand = await self.read_whm_attribute_expect_success(attribute="HeatDemand")
+        asserts.assert_equal(heatDemand, 0)
+
+        self.step("11b")
+        await self.check_whm_attribute("BoostState", Clusters.WaterHeaterManagement.Enums.BoostStateEnum.kInactive)
+
+        self.step("12")
+        await self.send_test_event_trigger_water_temperature20C_test_event()
+
+        self.step("12a")
+        heatDemand = await self.read_whm_attribute_expect_success(attribute="HeatDemand")
+        asserts.assert_equal(heatDemand, 0)
+
+        self.step("13")
+        await self.send_boost_command(duration=600)
+
+        self.step("13a")
+        heatDemand = await self.read_whm_attribute_expect_success(attribute="HeatDemand")
+        asserts.assert_not_equal(
+            heatDemand & Clusters.WaterHeaterManagement.Bitmaps.WaterHeaterDemandBitmap.kImmersionElement1, 0)
+
+        self.step("13b")
+        await self.check_whm_attribute("BoostState", Clusters.WaterHeaterManagement.Enums.BoostStateEnum.kActive)
+
+        self.step("14")
+        await self.send_test_event_trigger_water_temperature61C_test_event()
+
+        self.step("14a")
+        heatDemand = await self.read_whm_attribute_expect_success(attribute="HeatDemand")
+        asserts.assert_equal(heatDemand, 0)
+
+        self.step("14b")
+        await self.check_whm_attribute("BoostState", Clusters.WaterHeaterManagement.Enums.BoostStateEnum.kActive)
+
+        self.step("15")
+        await self.send_test_event_trigger_water_temperature20C_test_event()
+
+        self.step("15a")
+        heatDemand = await self.read_whm_attribute_expect_success(attribute="HeatDemand")
+        asserts.assert_not_equal(
```

Comment:
{valhas} greater than 0x00 (demand on at least one source) and (HeaterDemand & (!HeaterTypes)) is zero (demand is only from declared supported types)

Reply:
Done